features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/slowframes/DefaultUISlownessMetricDispatcherTest.kt

```diff
@@ -0,0 +1,215 @@
+        testedDispatcher.sendMetric(fakeViewId)
+
+        // Then
+        verify(internalLogger).logMetric(
+            argThat { invoke() == DefaultUISlownessMetricDispatcher.UI_SLOWNESS_MESSAGE },
+            eq(buildMetricAttributesMap(freezeFramesCount = 1, config = fakeSlowFramesConfiguration)),
+            eq(fakeSamplingRate),
+            eq(null)
+        )
+    }
+
+    @Test
+    fun `M increment ignoredFramesCount W incFreezeFrame`() {
+        // Given
+        testedDispatcher.onViewCreated(fakeViewId)
+
+        // When
+        testedDispatcher.incIgnoredFrame(fakeViewId)
+        testedDispatcher.sendMetric(fakeViewId)
+
+        // Then
+        verify(internalLogger).logMetric(
+            argThat { invoke() == DefaultUISlownessMetricDispatcher.UI_SLOWNESS_MESSAGE },
+            eq(buildMetricAttributesMap(ignoredFramesCount = 1, config = fakeSlowFramesConfiguration)),
+            eq(fakeSamplingRate),
+            eq(null)
+        )
+    }
+
+    @Test
+    fun `M send telemetry only once W sendSlowFramesTelemetry`() {
+        // Given
+        testedDispatcher.onViewCreated(fakeViewId)
+
+        // When
+        testedDispatcher.sendMetric(fakeViewId)
+        testedDispatcher.sendMetric(fakeViewId)
+
+        // Then
+        verify(internalLogger, times(1)).logMetric(
+            argThat { invoke() == DefaultUISlownessMetricDispatcher.UI_SLOWNESS_MESSAGE },
+            eq(buildMetricAttributesMap(config = fakeSlowFramesConfiguration)),
+            eq(fakeSamplingRate),
+            eq(null)
+        )
+    }
+
+    @Test
+    fun `M log warning W sendSlowFramesTelemetry twice`() {
+        // Given
+        testedDispatcher.onViewCreated(fakeViewId)
+
+        // When
+        testedDispatcher.sendMetric(fakeViewId)
+        testedDispatcher.sendMetric(fakeViewId)
+
+        // Then
+        verify(internalLogger).log(
+            level = eq(InternalLogger.Level.WARN),
+            target = eq(Target.TELEMETRY),
+            messageBuilder = argThat { invoke() == "No telemetry found for viewId=$fakeViewId" },
+            throwable = eq(null),
+            onlyOnce = eq(false),
+            additionalProperties = eq(null)
+        )
+    }
+
+    companion object {
+
+        private fun buildMetricAttributesMap(
+            slowFramesCount: Int = 0,
+            ignoredFramesCount: Int = 0,
+            freezeFramesCount: Int = 0,
+            config: SlowFramesConfiguration
+        ) = buildMetricAttributesMap(
+            slowFramesCount = slowFramesCount,
+            ignoredFramesCount = ignoredFramesCount,
+            freezeFramesCount = freezeFramesCount,
+            freezeDurationThreshold = config.freezeDurationThresholdNs,
+            maxSlowFramesAmount = config.maxSlowFramesAmount,
+            maxSlowFrameThresholdNs = config.maxSlowFrameThresholdNs,
+            minViewLifetimeThresholdNs = config.minViewLifetimeThresholdNs
+        )
+
+        private fun buildMetricAttributesMap(
```

Comment:
minor: this is a bit replicates what is in the code, so maybe instead we can assert individual values by pulling them from the map?